server/connection_handler.go

```diff
@@ -176,7 +177,9 @@ func (h *ConnectionHandler) setConn(conn net.Conn) {
 
 func (h *ConnectionHandler) handleStartup() error {
 	startupMessage, err := h.backend.ReceiveStartupMessage()
-	if err != nil {
+	if err == io.EOF {
```

Comment:
This doesn't seem correct.
https://www.postgresql.org/docs/15/protocol-flow.html
Once we receive the `*pgproto3.StartupMessage`, we have specific steps that need to be taken on our end. For now, since this doesn't include any user auth, we'll send the `ReadyForQuery`, which we're doing. So this `io.EOF` error only seems like it would come up if we're incorrectly calling `handleStartup` in a location that we shouldn't be.

Reply:
Under `SSLRequest`, we should update the error too since it was written for when we only denied SSL requests:
```
"error sending deny SSL request: %w"
```

Reply:
We’ve discussed this through direct messaging and have a way to proceed forward, so this PR is clear!